fswatcher/fswatcher.go

```diff
@@ -68,9 +73,22 @@ func (w *Sentry) Watch() (certCh <-chan tls.Certificate, errCh <-chan error) {
 			for eventsCh != nil && errorsCh != nil {
 				select {
 				case event, ok := <-eventsCh:
-					if !ok {
+					// Portions of this case are inspired by spf13/viper's WatchConfig.
+					// (c) 2014 Steve Francia. MIT Licensed.
+					currentPath, err := filepath.EvalSymlinks(certPath)
+					if err != nil {
+						w.errChan <- err
+					}
+
+					switch {
+					case !ok:
 						eventsCh = nil
-					} else if event.Op&fsnotify.Write == fsnotify.Write {
+					// 1. The event is for the certificate file and was a creation or modification event
+					case filepath.Clean(event.Name) == certPath && (event.Op&fsnotify.Create != 0 || event.Op&fsnotify.Write != 0),
```

Comment:
I think it might be easier to follow the logic if you split it into multiple case statements:
```go
// There are 2 separate cases to handle which would trigger a load:
switch {
case !ok:
	eventsCh = nil
case filepath.Clean(event.Name) == certPath && (event.Op&fsnotify.Create != 0 || event.Op&fsnotify.Write != 0):
	// 1. The event is for the certificate file and was a creation or modification event
	w.loadCertificate()
case currentPath != "" && currentPath != realCertPath:
	// 2. We observed the symlink has changed (eg, in Kubernetes)
	realCertPath = currentPath
	w.loadCertificate()
}
```

Reply:
This doesn't seem that much easier for me.

Reply:
I found it hard to follow the case statement split over multiple lines with a comma. Everyone's mileage may vary.

Reply:
It took me 3 readings to find the comma and interpret it correctly, even after seeing this comment. I think breaking this up into 2 cases makes more sense and is easier to follow.